### PR TITLE
Add functions.json

### PR DESCRIPTION
**Pull request by martindekov**
Adding function.json file as single
file which will be storing the functions
and appending platform field to determine
which function runs on which platform

References  Issue [#1099](https://github.com/openfaas/faas/issues/1099)

Signed-off-by: mdekov <mdekov@vmware.com>

### CODE REVIEW

**Comment by rgee0** [comment]
This needs some thought.  I'm not sure about the repetition.  I wrote about a way to achieve this in https://github.com/openfaas/faas/issues/429.  By using a map we have the list of supported platforms as keys and the respective image as values, which means the remaining blurb remains the same.  I'm not too concerned about the GitHub repo as for the most part these should be same - although a different person might build the image, the link should be to the original repo.

It'd be interesting to get some wider thoughts on this.

**Comment by alexellis** [comment]
What repetition @rgee0 ?

Alex

**Comment by rgee0** [comment]
NodeInfo, for example, appears 3 times, once for each platform. Everything bar the platform and image should be the same.

**Comment by martindekov** [comment]
Thanks @rgee0 and @viveksyngh for taking a look. Richard for your suggestion should we go with something like:
```json
    {
        "title": "NodeInfo",
        "description": "Get info about the machine that you're deployed on. Tells CPU count, hostname, OS, and Uptime",
        "images": {
                "arm64" : "functions/nodeinfo:arm64-latest",
                "armhf" : "functions/nodeinfo:armhf-latest",
                "x86_64": "functions/nodeinfo:latest"
        },
        "name": "nodeinfo",
        "fprocess": "node main.js",
        "network": "func_functions",
        "repo_url": "https://github.com/openfaas/faas/tree/master/sample-functions/NodeInfo"
    }
}
```
or append `platform` like:
```json
    {
        "title": "NodeInfo",
        "description": "Get info about the machine that you're deployed on. Tells CPU count, hostname, OS, and Uptime",
        "images": {
                 "platform": {
                        "arm64" : "functions/nodeinfo:arm64-latest",
                        "armhf" : "functions/nodeinfo:armhf-latest",
                        "x86_64": "functions/nodeinfo:latest"
                 }
        },
        "name": "nodeinfo",
        "fprocess": "node main.js",
        "network": "func_functions",
        "repo_url": "https://github.com/openfaas/faas/tree/master/sample-functions/NodeInfo"
    }
}
```

**Comment by pcorbel** [comment]
@martindekov I think we should keep it simple and stupid, i.e suggestion 2
From my point of view, the "platform" key is not very useful

**Comment by martindekov** [comment]
Yeah thanks for the suggestion @pcorbel I will go without platform.

File: functions.json [hunk 1]
```diff
@@ -2,81 +2,57 @@
...
-    {
-        "title": "Left-Pad",
-        "description": "left-pad on OpenFaaS",
-        "image": "rgee0/faas-leftpad",
-        "name": "leftpad",
-        "repo_url": "https://github.com/faas-and-furious/faas-leftpad",
-        "platform": "armhf"
-    },
-    {
-        "title": "Manifest Query",
```
Comment (martindekov) [inline functions.json]: Not sure why this is in [store-armhf.json](https://github.com/openfaas/store/blob/9e492dfd956adaf499449315542b996c2a452f21/store-armhf.json#L26) and [store.json](https://github.com/openfaas/store/blob/9e492dfd956adaf499449315542b996c2a452f21/store.json#L185) when the images are the same `rgee0/of-mquery:1.2` so I presumed `x86_64` architecture

Comment (rgee0) [reply functions.json]: It's a multi-arch image - works on both

Comment (martindekov) [reply functions.json]: Ok thanks for the input @rgee0 I think this is ready.

**Comment by martindekov** [comment]
I think the PR is ready. If you don't have any other suggestions I will squash the commits

**Comment by martindekov** [comment]
Done @rgee0 the field had different intent initially.

**Pull request merged**